internal/runners/prepare/prepare.go

```diff
@@ -1,30 +1,63 @@
 package prepare
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
 	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/internal/output"
 	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/subshell"
 )
 
 type primeable interface {
 	primer.Outputer
+	primer.Subsheller
 }
 
 // Prepare manages the prepare execution context.
 type Prepare struct {
-	out output.Outputer
+	out      output.Outputer
+	subshell subshell.SubShell
 }
 
 // New prepares a prepare execution context for use.
 func New(prime primeable) *Prepare {
 	return &Prepare{
-		out: prime.Output(),
+		out:      prime.Output(),
+		subshell: prime.Subshell(),
 	}
 }
 
 // Run executes the prepare behavior.
 func (r *Prepare) Run() error {
 	logging.Debug("ExecutePrepare")
 
+	binDir := filepath.Join(config.CachePath(), "prepareBin")
+	fail := fileutils.Mkdir(binDir)
+	if fail != nil {
+		return locale.WrapError(fail.ToError(), "err_prepare_bin_dir", "Could not create global installations directory")
+	}
+
+	envUpdates := map[string]string{
+		"PATH": fmt.Sprintf("%s%s%s", binDir, string(os.PathListSeparator), os.Getenv("PATH")),
+	}
+
+	fail = r.subshell.WriteUserEnv(envUpdates, true)
```

Comment:
There's an issue with the current approach. `WriteUserEnv` has logic that cleans up previous writes to the env before writing the new one. The problem here is that it assumes specifically the `deploy` use-case. Which means that if you use both `state deploy` and `state activate --default` then one will override the other.

We need to update the mechanic here so the two can live side by side.